src/DotLiquid/StandardFilters.cs

```diff
@@ -931,6 +931,53 @@ public static IEnumerable Compact(object input)
             ary.RemoveAll(item => item == null);
             return ary;
         }
+
+        /// <summary>
+        /// Creates an array including only the objects with a given property value, or any truthy value by default.
+        /// </summary>
+        /// <param name="input">an array to be filtered</param>
+        /// <param name="property">The name of the property to filter by</param>
+        /// <param name="target_value">Value to retain, if null object containing this property are retained</param>
+        /// <returns></returns>
+        public static IEnumerable Where(object input, string property, string target_value = null)
+        {
+            if (input == null)
+                return null;
+
+            if (property.IsNullOrWhiteSpace())
+                throw new ArgumentNullException(paramName: nameof(property), message: $"'{nameof(property)}' cannot be null or empty.");
+
+            List<object> inputList = (input is IEnumerable enumerableInput) ? enumerableInput.Cast<object>().ToList() : null;
+
+            return inputList.Where(s => s.Matches(property, target_value));
+        }
+
+        private static bool Matches(this object any, string property, string target_value)
+        {
+            bool matches = false;
+            if (any is IDictionary dictionary && dictionary.Contains(property))
+            {
+                // the 'any' dictionary has the filter property
+                var propertyValue = dictionary[property];
+                if (target_value == null)
+                {
+                    // If no filter value provided, check the property value is truthy.
+                    matches = propertyValue != null && !false.Equals(propertyValue);
+                }
+                else
+                {
+                    matches = target_value.Equals(propertyValue);
+                }
+            }
+            else if (any.RespondTo(property))

```

Comment:
Can any be null? i.e. can there be a null entry in the input list?

Reply:
Test case and null-check added.